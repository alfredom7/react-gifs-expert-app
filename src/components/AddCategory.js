import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // ''

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if ( inputValue.trim().length > 2 ) {
            setCategories( cats => [ inputValue ] );
            setInputValue('');
        }

    }

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <div class="input-group mb-3">
            <input 
                type="text" 
                value={ inputValue } 
                onChange={ handleInputChange } 
                className="form-control" 
                placeholder="Write for search..." 
                aria-label="Write for search..." 
                aria-describedby="button-addon2"
            />
            <button onClick={ handleSubmit } className="btn btn-success" type="button" id="button-addon2">Search</button>
            </div>
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
