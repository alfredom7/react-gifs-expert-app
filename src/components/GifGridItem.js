import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url }) => {

    return (
        <div className='col-md-4 col-lg-3 col-xl-2 pb-3'>
            <div className="mycard card text-white bg-dark animate__animated animate__fadeIn shadow-sm rounded-3">
            <img src={ url } alt={ title } className="card-img-top img-grid"/>
            <div className="card-body text-center">
                <p className="card-title img-my-title text-capitalize text-truncate">{ title }</p>
                <a href="#" className="btn btn-success btn-sm">Select</a>
            </div>
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}