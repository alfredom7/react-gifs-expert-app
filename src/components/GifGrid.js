import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );

    return (
        <>
            <h6 className="animate__animated animate__fadeIn pb-2 resultsfor">Results for: { category } </h6>

            { loading && <p className="animate__animated animate__flash">Loading</p> }

            <div className="card-grid row">
                
                {
                    images.map( img => (
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
            
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
