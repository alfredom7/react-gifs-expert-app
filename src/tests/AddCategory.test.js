
import React from 'react';
import { shallow } from 'enzyme'
import { AddCategory } from '../components/AddCategory';

describe('Pruebas en el <AddCategory/>', () => {

  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
  
  beforeEach(()=>{
    wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
  })
  
  test('Deberia tomar una snapshot del componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Deberia devolver el texto enviado como argumento en la funcion', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';

    input.simulate('change', { target: { value }})
    expect(wrapper.find('p').text().trim()).toBe(value)
    
  });

  test('No debe postear la informaciÃ³n con submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect( setCategories ).not.toHaveBeenCalled();
  });

  test('Debe llamar el setCategories y limpiar la caja de texto', () => {
    const value = 'Hola mundo';
    // 1. Simular el inputChange
    wrapper.find('input').simulate('change', { target: { value }})
    // 2. Simular el submit
    wrapper.find('form').simulate('submit', { preventDefault(){} });

    // 3. setCategories debe haber sido llamado
    expect(setCategories).toHaveBeenCalled()
    expect(setCategories).toHaveBeenCalledWith( expect.any(Function) )
    

    // 4. El valor del input debe estar en ''
    expect(wrapper.find('input').prop('value')).toBe('')
    
  });

});