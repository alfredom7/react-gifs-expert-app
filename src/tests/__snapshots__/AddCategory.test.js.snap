// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <AddCategory/> Deberia tomar una snapshot del componente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <div
    class="input-group mb-3"
  >
    <input
      aria-describedby="button-addon2"
      aria-label="Write for search..."
      className="form-control"
      onChange={[Function]}
      placeholder="Write for search..."
      type="text"
      value=""
    />
    <button
      className="btn btn-success"
      id="button-addon2"
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </div>
</form>
`;
